### ISSUE TITLE
GTK/Qt theme not working

### ISSUE THREAD

**Issue by musjj** [NixOS/nixpkgs#244895]
### Describe the bug
I've followed the instructions described in the [manual](https://nixos.org/manual/nixos/unstable/#sec-x11-gtk-and-qt-themes), but Qt applications are not using my GTK theme.

### Steps To Reproduce
```nix
qt.enable = true;
qt.platformTheme = "gtk2";
qt.style = "gtk2";
environment.systemPackages = [ pkgs.qalculate-qt ];
```

Running `qalculate-qt` will yield the following error:

```bash
QApplication: invalid style override 'gtk2' passed, ignoring it.
	Available styles: Windows, Fusion
```

### Expected behavior
Qt applications should properly detect the `gtk2` style.

### Notify maintainers

@romildo 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.39, NixOS, 23.11 (Tapir), 23.11.20230721.9ca7856`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.1`
 - nixpkgs: `/nix/store/d1fspdh09fpk7dkl5s667vhpkgv12dsx-source`
```

**Comment by bobby285271** [comment]
Yeah qalculate-qt (and anki) are Qt 6 applications, I expect the theming not working before someone package https://github.com/trialuser02/qt6gtk2 (which is probably the reason the manual says "Qt 5 applications" I guess).

**Comment by Misterio77** [comment]
Here's qt6gtk2: https://github.com/NixOS/nixpkgs/issues/244895

With that PR merged, this issue should be fixed right out of the box when using NixOS to manage qt.

**Comment by musjj** [comment]
Thank you for the PR!
